core/src/main/java/com/datastrato/gravitino/storage/relational/mysql/MySQLBackend.java

```diff
@@ -25,41 +36,160 @@ public class MySQLBackend implements RelationalBackend {
 
   /** Initialize the MySQL backend instance. */
   @Override
-  public void initialize(Config config) {}
+  public void initialize(Config config) {
+    SqlSessionFactoryHelper.getInstance().init(config);
+  }
 
   @Override
   public <E extends Entity & HasIdentifier> List<E> list(
-      Namespace namespace, Entity.EntityType entityType) throws NoSuchEntityException {
-    throw new UnsupportedOperationException("Unsupported operation now.");
+      Namespace namespace, Entity.EntityType entityType) {
+    if (entityType == Entity.EntityType.METALAKE) {
+      List<MetalakePO> metalakePOS =
+          SessionUtils.getSessionWithFinish(
+              MetalakeMetaMapper.class, mapper -> mapper.listMetalakePOs());
+      return metalakePOS != null
+          ? (List<E>) POConverters.fromMetalakePOs(metalakePOS)
+          : new ArrayList<>();
+    } else {
+      throw new IllegalArgumentException(
+          String.format("Unsupported entity type: %s for list operation", entityType));
+    }
   }
 
   @Override
   public boolean exists(NameIdentifier ident, Entity.EntityType entityType) {
-    throw new UnsupportedOperationException("Unsupported operation now.");
+    if (entityType == Entity.EntityType.METALAKE) {
+      MetalakePO metalakePO =
+          SessionUtils.getSessionWithFinish(
+              MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeMetaByName(ident.name()));
+      return metalakePO != null;
+    } else {
+      throw new IllegalArgumentException(
+          String.format("Unsupported entity type: %s for exists operation", entityType));
+    }
   }
 
   @Override
   public <E extends Entity & HasIdentifier> void insert(E e, boolean overwritten)
       throws EntityAlreadyExistsException {
-    throw new UnsupportedOperationException("Unsupported operation now.");
+    if (e instanceof BaseMetalake) {
+      MetalakePO metalakePO =
+          SessionUtils.getSessionWithFinish(
+              MetalakeMetaMapper.class,
+              mapper -> mapper.selectMetalakeMetaByName(e.nameIdentifier().name()));
+      if (!overwritten && metalakePO != null) {
+        throw new EntityAlreadyExistsException(
+            String.format("Metalake entity: %s already exists", e.nameIdentifier().name()));
+      }
+      SessionUtils.doSessionWithFinish(
+          MetalakeMetaMapper.class,
+          mapper -> {
+            if (overwritten) {
+              try {
+                mapper.insertMetalakeMetaWithUpdate(POConverters.toMetalakePO((BaseMetalake) e));
+              } catch (JsonProcessingException ex) {
+                throw new RuntimeException(ex);
+              }
+            } else {
+              try {
+                mapper.insertMetalakeMeta(POConverters.toMetalakePO((BaseMetalake) e));
+              } catch (JsonProcessingException ex) {
+                throw new RuntimeException(ex);
+              }
+            }
```

Comment:
```suggestion
            try {
              if (overwritten) {
                mapper.insertMetalakeMetaWithUpdate(POConverters.toMetalakePO((BaseMetalake) e));
              } else {
                mapper.insertMetalakeMeta(POConverters.toMetalakePO((BaseMetalake) e));
              }
            } catch (JsonProcessingException ex) {
              throw new RuntimeException(ex);
            }
```

Reply:
Prefer POConverter handling the JsonProcessingException.